### PR TITLE
Improves magic handling

### PR DESCRIPTION
**Pull request by emeryberger**
Fixes https://github.com/plasma-umass/scalene/issues/843 by providing more robust magic detection and removal.

### CODE REVIEW

File: scalene/scalene_analysis.py [hunk 1]
```diff
@@ -206,10 +207,37 @@ def walk(
...
+        for line in lines:
+            stripped_line = line.strip()
+
+            # Check if we're entering or exiting a multi-line comment or docstring
+            if stripped_line.startswith(('"""', "'''")):
+                if in_multiline_comment:
+                    if stripped_line.endswith(('"""', "'''")) and len(stripped_line) > 3:
+                        in_multiline_comment = False
+                    else:
+                        in_multiline_comment = False
```
Comment (D-VR) [inline scalene/scalene_analysis.py:227]: why is this line false?

**Comment by D-VR** [comment]
trying out this branch with the previous test environment leads to consistent crashes using the below command (although it also succeeded once).

`scalene --profile-all --profile-exclude threading.py,dataclasses.py --stacks --cli --cpu --memory --json --outfile test.json --cpu-sampling-rate 0.001 test.py --- -h`

Below you can see the two crash types I've encountered, the key 2785 being more common:

```
(venv) ➜  improvepal git:(main) ✗ scalene --profile-all --profile-exclude threading.py,dataclasses.py --stacks --cli --cpu --memory --json --outfile test.json --cpu-sampling-rate 0.001 test.py --- -h
<cmd output>
Scalene: An exception of type KeyError occurred. Arguments:
(2785,)
Traceback (most recent call last):
  File "user/venv/lib/python3.12/site-packages/scalene/scalene_profiler.py", line 2153, in run_profiler
    exit_status = profiler.profile_code(
                  ^^^^^^^^^^^^^^^^^^^^^^
  File "user/venv/lib/python3.12/site-packages/scalene/scalene_profiler.py", line 1816, in profile_code
    did_output = Scalene.output_profile(left)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "user/venv/lib/python3.12/site-packages/scalene/scalene_profiler.py", line 871, in output_profile
    json_output = Scalene.__json.output_profiles(
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "user/venv/lib/python3.12/site-packages/scalene/scalene_json.py", line 513, in output_profiles
    profile_line["start_region_line"] = enclosing_regions[
                                        ^^^^^^^^^^^^^^^^^^
KeyError: 2785
```

```
(venv) ➜  improvepal git:(main) ✗ scalene --profile-all --profile-exclude threading.py,dataclasses.py --stacks --cli --cpu --memory --json --outfile test.json --cpu-sampling-rate 0.001 test.py --- -h
<cmd output>
Scalene: An exception of type SyntaxError occurred. Arguments:
('invalid syntax', ('<unknown>', 1100, 3, '  def _is_in_control_flow(self, op):\n', 1100, 6))
Traceback (most recent call last):
  File "user/venv/lib/python3.12/site-packages/scalene/scalene_profiler.py", line 2153, in run_profiler
    exit_status = profiler.profile_code(
                  ^^^^^^^^^^^^^^^^^^^^^^
  File "user/venv/lib/python3.12/site-packages/scalene/scalene_profiler.py", line 1816, in profile_code
    did_output = Scalene.output_profile(left)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "user/venv/lib/python3.12/site-packages/scalene/scalene_profiler.py", line 871, in output_profile
    json_output = Scalene.__json.output_profiles(
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "user/venv/lib/python3.12/site-packages/scalene/scalene_json.py", line 486, in output_profiles
    enclosing_regions = ScaleneAnalysis.find_regions(code_str)
                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "user/venv/lib/python3.12/site-packages/scalene/scalene_analysis.py", line 110, in find_regions
    tree = ast.parse(src)
           ^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/ast.py", line 52, in parse
    return compile(source, filename, mode, flags,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<unknown>", line 1100
    def _is_in_control_flow(self, op):
    ^^^
SyntaxError: invalid syntax
```
removing the entire function leads to no crashes.

File: scalene/scalene_analysis.py (hunk 1 above)
Comment (emeryberger) [reply scalene/scalene_analysis.py:227]: Yes, will revisit.

**Comment by emeryberger** [comment]
Trying a different approach: https://github.com/plasma-umass/scalene/pull/847.

**Pull request closed without merging**